In [1]:
import pandas as pd
import numpy as np

In [2]:
# Load in the dataframe from project part 1 as basics:
basics = pd.read_csv('Data/basics.csv.gz')
basics.head()

tconst titleType                                       primaryTitle  \
0  tt0035423     movie                                     Kate & Leopold   
1  tt0062336     movie  The Tango of the Widower and Its Distorting Mi...   
2  tt0069049     movie                         The Other Side of the Wind   
3  tt0088751     movie                                  The Naked Monster   
4  tt0096056     movie                               Crime and Punishment   

                               originalTitle  isAdult  startYear  endYear  \
0                             Kate & Leopold        0       2001      NaN   
1  El Tango del Viudo y Su Espejo Deformante        0       2020      NaN   
2                 The Other Side of the Wind        0       2018      NaN   
3                          The Naked Monster        0       2005      NaN   
4                       Crime and Punishment        0       2002      NaN   

   runtimeMinutes                  genres  
0             118  Comedy,Fantasy,Romance  
1              70                   Drama  
2             122                   Drama  
3             100    Comedy,Horror,Sci-Fi  
4             126                   Drama

In [3]:
print('Titles in basics file from 2000: ' + str(len(basics[basics['startYear'] == 2000])))
print('Titles in basics file from 2001: ' + str(len(basics[basics['startYear'] == 2001])))

Titles in basics file from 2000: 1415
Titles in basics file from 2001: 1532


In [4]:
y1 = pd.read_csv('Data/final_tmdb_data_2000.csv.gz')
y1.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres  \
0         NaN                                                NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   

                               homepage        id original_language  \
0                                   NaN       NaN               NaN   
1                                   NaN   62127.0                en   
2                                   NaN  110977.0                en   
3                                   NaN  442869.0                hi   
4  http://www.wkw-inthemoodforlove.com/     843.0                cn   

    original_title  ...     revenue  runtime  \
0              NaN  ...         NaN      NaN   
1  The Fantasticks  ...         0.0     86.0   
2    For the Cause  ...         0.0    100.0   
3             Gang  ...         0.0    152.0   
4             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         4.000        1.0           NaN  
4    0.0         8.104     2027.0            PG  

[5 rows x 26 columns]

In [5]:
y2 = pd.read_csv('Data/final_tmdb_data_2001.csv.gz')
y2.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0035423    0.0  /hfeiSfWYujh6MKhtGTXyK3DD4nN.jpg                   NaN   
2  tt0114447    0.0                               NaN                   NaN   
3  tt0118589    0.0  /9NZAirJahVilTiDNCHLFcdkwkiy.jpg                   NaN   
4  tt0118652    0.0  /mWxJEFRMvkG4UItYJkRDMgWQ08Y.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  48000000.0  [{'id': 10749, 'name': 'Romance'}, {'id': 14, ...      NaN   
2         0.0  [{'id': 53, 'name': 'Thriller'}, {'id': 28, 'n...      NaN   
3  22000000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...      NaN   
4   1000000.0  [{'id': 27, 'name': 'Horror'}, {'id': 9648, 'n...      NaN   

         id original_language         original_title  ...     revenue  \
0       NaN               NaN                    NaN  ...         NaN   
1   11232.0                en         Kate & Leopold  ...  76019048.0   
2  151007.0                en       The Silent Force  ...         0.0   
3   10696.0                en                Glitter  ...   5271666.0   
4   17140.0                en  The Attic Expeditions  ...         0.0   

   runtime                                   spoken_languages    status  \
0      NaN                                                NaN       NaN   
1    118.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2     90.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3    104.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
4    100.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline                  title  \
0                                                NaN                    NaN   
1  If they lived in the same century, they'd be p...         Kate & Leopold   
2  They left him for dead... They should have fin...       The Silent Force   
3   In music she found her dream, her love, herself.                Glitter   
4  His search for peace of mind... will leave his...  The Attic Expeditions   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         6.324     1155.0         PG-13  
2    0.0         5.000        3.0           NaN  
3    0.0         4.600      118.0         PG-13  
4    0.0         5.100       28.0             R  

[5 rows x 26 columns]

In [6]:
frames = [y1, y2]
merged = pd.concat(frames)

In [7]:
y1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1211 entries, 0 to 1210
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                1211 non-null   object 
 1   adult                  1210 non-null   float64
 2   backdrop_path          649 non-null    object 
 3   belongs_to_collection  108 non-null    object 
 4   budget                 1210 non-null   float64
 5   genres                 1210 non-null   object 
 6   homepage               68 non-null     object 
 7   id                     1210 non-null   float64
 8   original_language      1210 non-null   object 
 9   original_title         1210 non-null   object 
 10  overview               1188 non-null   object 
 11  popularity             1210 non-null   float64
 12  poster_path            1079 non-null   object 
 13  production_companies   1210 non-null   object 
 14  production_countries   1210 non-null   object 
 15  rele

In [8]:
y2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1293 entries, 0 to 1292
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                1293 non-null   object 
 1   adult                  1292 non-null   float64
 2   backdrop_path          668 non-null    object 
 3   belongs_to_collection  87 non-null     object 
 4   budget                 1292 non-null   float64
 5   genres                 1292 non-null   object 
 6   homepage               107 non-null    object 
 7   id                     1292 non-null   float64
 8   original_language      1292 non-null   object 
 9   original_title         1292 non-null   object 
 10  overview               1264 non-null   object 
 11  popularity             1292 non-null   float64
 12  poster_path            1143 non-null   object 
 13  production_companies   1292 non-null   object 
 14  production_countries   1292 non-null   object 
 15  rele

In [9]:
merged.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2504 entries, 0 to 1292
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                2504 non-null   object 
 1   adult                  2502 non-null   float64
 2   backdrop_path          1317 non-null   object 
 3   belongs_to_collection  195 non-null    object 
 4   budget                 2502 non-null   float64
 5   genres                 2502 non-null   object 
 6   homepage               175 non-null    object 
 7   id                     2502 non-null   float64
 8   original_language      2502 non-null   object 
 9   original_title         2502 non-null   object 
 10  overview               2452 non-null   object 
 11  popularity             2502 non-null   float64
 12  poster_path            2222 non-null   object 
 13  production_companies   2502 non-null   object 
 14  production_countries   2502 non-null   object 
 15  rele

Total number of records in merged dataframe is reasonable given number of 2000/2001 titles in the basics file along with the number of errors thrown during the API calls.

In [27]:
merged_with_financials = merged[(merged['budget'] > 0) | (merged['revenue'] > 0)]

In [29]:
len(merged_with_financials)

625

625 movies with some valid financial data.

How many movies are there in each of the certification categories (G/PG/PG-13/R)?

In [30]:
merged_with_financials['certification'].value_counts()

R        228
PG-13    130
PG        36
NR        17
G         14
Name: certification, dtype: int64

What is the average revenue per certification category?

In [42]:
pd.set_option('display.float_format', '${:,.0f}'.format)
merged_with_financials.groupby('certification')['revenue'].mean()

certification
G       $123,746,275
NR        $8,887,619
PG      $109,218,395
PG-13   $100,026,552
R        $33,351,745
Name: revenue, dtype: float64

What is the average budget per certification category?

In [43]:
merged_with_financials.groupby('certification')['budget'].mean()

certification
G       $40,857,143
NR       $6,300,006
PG      $43,729,167
PG-13   $43,483,907
R       $19,822,388
Name: budget, dtype: float64

In [44]:
merged.to_csv("Data/tmdb_results_combined.csv.gz", compression="gzip", index=False)